presto-main/src/main/java/io/prestosql/operator/JoinProbe.java

```diff
@@ -118,8 +118,22 @@ private void fillJoinPositionCache(LookupSource lookupSource)
         // Extracted to local variables for performance reasons
         int firstPosition = this.position & (~JOIN_POSITIONS_CACHE_MASK);
         int limit = min(JOIN_POSITIONS_CACHE_SIZE, positionCount - firstPosition);
-        for (int i = 0; i < limit; ++i) {
-            joinPositionsCache[i] = getJoinPosition(firstPosition + i, lookupSource);
+        if (mayContainNullRows()) {
+            for (int i = 0; i < limit; ++i) {
+                joinPositionsCache[i] = getJoinPosition(firstPosition + i, lookupSource);
+            }
+        }
+        else {
+            if (probeHashBlock.isPresent()) {
+                long[] rawHashes = new long[limit];
```

Comment:
does caching `rawHashes` array makes a difference?

Reply:
It's for a single-time use. I don't see a reason to cache it.